assets/app/view/game/buy_company_from_other_player.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module View
+  module Game
+    class BuyCompanyFromOtherPlayer < Snabberb::Component
+      include Actionable
+
+      needs :game, store: true
+      needs :selected_companies, default: [], store: true
+      needs :selected_company, default: nil, store: true
+
+      def render
+        @step = @game.round.active_step
+        @current_actions = @step.current_actions
+        return h(:div, {}, '') if !@step.can_buy_any?(@game.current_entity) || !@current_actions.include?('buy_company')
+
+        buyable_companies = @game.companies.select { |c| purchasable?(c) }
+        return h(:div, {}, '') if buyable_companies.empty?
+
+        props = {
+          style: {
+            grid: '1fr / 5rem 4rem',
+            alignItems: 'center',
+            margin: '1rem 0 1.5rem 0',
+          },
+        }
+
+        input = h(:input, style: { marginBottom: '0.5rem', marginRight: '1rem' }, props: {
+          value: 1,
+          type: 'number',
+          min: 1,
+          max: @game.current_entity.cash.to_s.size,
+          size: @game.current_entity.cash.to_s.size,
+        })
+        children = [input]
+        buyable_companies
+          .select { |c| purchasable?(c) }
```

Comment:
why are you doing select purchasable twice?

Reply:
Ops. Fixed